app/models/mixins/vim_connect_mixin.rb

```diff
@@ -42,4 +41,38 @@ def with_provider_connection(options = {})
       vim.try(:disconnect) rescue nil
     end
   end
+
+  module ClassMethods
+    def make_connection(options)
+      if options[:fault_tolerant]
+        MiqFaultTolerantVim.new(options)
+      else
+        MiqVim.new(options[:ip], options[:user],options[:pass])
+      end
+    end
+
+    def raw_connect(options)
```

Comment:
raw_connect for the other providers typically does not verify the connection.  It's purpose is closer to your make_connection method.  connect and validate are usually implemented by calling raw_connect to get the native handle.